guides/md/keras-tuner/distributed_tuning.md

```diff
@@ -0,0 +1,170 @@
+
+### Configuring distributed mode
+
+Configuring distributed mode for Keras Tuner only requires setting three environment variables:
+
+**KERASTUNER_TUNER_ID**: This should be set to "chief" for the chief process. Other workers should be passed a unique ID (by convention, "tuner0", "tuner1", etc).
+
+**KERASTUNER_ORACLE_IP**: The IP address or hostname that the chief service should run on. All workers should be able to resolve and access this address.
+
+**KERASTUNER_ORACLE_PORT**: The port that the chief service should run on. This can be freely chosen, but must be a port that is accessible to the other workers. Instances communicate via the [gRPC](https://www.grpc.io) protocol.
+
+The same code can be run on all workers. Additional considerations for distributed mode are:
+
+- All workers should have access to a centralized file system to which they can write their results.
+- All workers should be able to access the necessary training and validation data needed for tuning.
+- To support fault-tolerance, `overwrite` should be kept as `False` in `Tuner.__init__` (`False` is the default).
+
+Example bash script for chief service (sample code for `run_tuning.py` at bottom of page):
+
+
+```python
+!!export KERASTUNER_TUNER_ID="chief"
+!!export KERASTUNER_ORACLE_IP="127.0.0.1"
+!!export KERASTUNER_ORACLE_PORT="8000"
+!!python run_tuning.py
+```
+
+
+
+
+<div class="k-default-codeblock">
+```
+["python: can't open file 'run_tuning.py': [Errno 2] No such file or directory"]
+
+```
+</div>
+Example bash script for worker:
+
+
+```python
+!export KERASTUNER_TUNER_ID="tuner0"
+!export KERASTUNER_ORACLE_IP="127.0.0.1"
+!export KERASTUNER_ORACLE_PORT="8000"
+!python run_tuning.py
+```
+
+<div class="k-default-codeblock">
+```
+python: can't open file 'run_tuning.py': [Errno 2] No such file or directory
+
+```
+</div>
+### Data parallelism with tf.distribute
+
+Keras Tuner also supports data parallelism via [tf.distribute](https://www.tensorflow.org/tutorials/distribute/keras). Data parallelism and distributed tuning can be combined. For example, if you have 10 workers with 4 GPUs on each worker, you can run 10 parallel trials with each trial training on 4 GPUs by using [tf.distribute.MirroredStrategy](https://www.tensorflow.org/api_docs/python/tf/distribute/MirroredStrategy). You can also run each trial on TPUs via [tf.distribute.experimental.TPUStrategy](https://www.tensorflow.org/api_docs/python/tf/distribute/experimental/TPUStrategy). Currently [tf.distribute.MultiWorkerMirroredStrategy](https://www.tensorflow.org/api_docs/python/tf/distribute/experimental/MultiWorkerMirroredStrategy) is not supported, but support for this is on the roadmap.
+
+
+### Example code
+
+When the enviroment variables described above are set, the example below will run distributed tuning and will also use data parallelism within each trial via `tf.distribute`. The example loads MNIST from `tensorflow_datasets` and uses hyperband for the hyperparameter search.
```

Comment:
Suggestions:

- "will... will.." -> "will run.. and use data parallelism..."
- Hyperband (capitalized) + link to https://arxiv.org/pdf/1603.06560.pdf

```suggestion
When the enviroment variables described above are set, the example below will run distributed tuning and will also use data parallelism within each trial via `tf.distribute`. The example loads MNIST from `tensorflow_datasets` and uses [Hyperband](https://arxiv.org/pdf/1603.06560.pdf) for the hyperparameter search.
```